src/gui/IconDownloader.cpp

```diff
@@ -73,16 +73,24 @@ namespace
 void IconDownloader::setUrl(const QString& entryUrl)
 {
     m_url = entryUrl;
-    QUrl url(m_url);
-    if (!url.isValid()) {
+    QUrl url = QUrl::fromUserInput(m_url);
+    if (!url.isValid() || url.host().isEmpty()) {
         return;
     }
 
     m_redirects = 0;
     m_urlsToTry.clear();
 
+    // If no scheme is specified, fall back to http as http is usually redirected to https, but not vice versa
     if (url.scheme().isEmpty()) {
-        url.setUrl(QString("https://%1").arg(url.toString()));
+        url.setScheme("http");
```

Comment:
Any specific reason for defaulting to http? Generally we default to https.

Reply:
I have defaulted to http:// here as that maximizes the probability of finding the correct service. However i agree that it would be better to default to HTTPS, especially given that the user may specify sensitive data as part of the url. I will update the PR as soon as the other issue has been discussed.